syntax-markdown-flavor/syntax-markdown-flavor-github/src/main/java/org/xwiki/contrib/rendering/markdown/flavor/github/MarkdownGithubConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.rendering.markdown.flavor.github;
+
+import com.vladsch.flexmark.util.data.MutableDataHolder;
+import org.xwiki.component.annotation.Role;
+
+/**
+ * Configuration for GitHub Flavored Markdown.
+ *
+ * @version $Id$
+ * @since 8.7
+ */
+@Role
+public interface MarkdownGithubConfiguration
```

Comment:
That's still not good: you don't need a specific Component Role with various implementation for the configuration of Markdown Github Flavor. You need a Component *Role* for various implementation of the configuration of Markdown Flavors. One of the flavor is the actual standard Markdown implementation which already exist, and another one is the Github flavor that you create. 

So basically you need to edit the already existing role for MarkdownConfiguration (it will be the only Component Role or interface to be used everywhere) to add indeed the `getOptions` method (or maybe something else more specific to ParserEmultationProfile, but we can check later).
Then you have the Default implementation which already exist, and you will provide a new configuration for Github.